Objects/complexobject.c

```diff
@@ -174,23 +232,31 @@ _Py_c_pow(Py_complex a, Py_complex b)
         r.real = len*cos(phase);
         r.imag = len*sin(phase);
 
-        _Py_ADJUST_ERANGE2(r.real, r.imag);
+        if (isfinite(a.real) && isfinite(a.imag)
+            && isfinite(b.real) && isfinite(b.imag))
+        {
+            _Py_ADJUST_ERANGE2(r.real, r.imag);
+        }
```

Comment:
Without this, `complex(inf)**101` raised an OverflowError.

There is no test for this, because the result (`(inf+nanj)`) is still incorrect -- it should be `(inf+0j)`, as for smaller exponents.